pkg/frontend/adminactions/resources_list.go

```diff
@@ -44,35 +59,55 @@ func (a *azureActions) ResourcesList(ctx context.Context) ([]byte, error) {
 			a.log.Warnf("API version not found for type %q", *res.Type)
 			continue
 		}
+
+		if hasWritten {
+			writer.Write(comaAsByteArray)
+		}
+		hasWritten = true
+
 		switch *res.Type {
 		case "Microsoft.Compute/virtualMachines":
 			vm, err := a.virtualMachines.Get(ctx, clusterRGName, *res.Name, mgmtcompute.InstanceView)
 			if err != nil {
 				a.log.Warn(err) // can happen when the ARM cache is lagging
-				armResources = append(armResources, arm.Resource{
+				a.writeObject(writer, arm.Resource{
 					Resource: res,
 				})
 				continue
 			}
-			armResources = append(armResources, arm.Resource{
+			a.writeObject(writer, arm.Resource{
 				Resource: vm,
 			})
 		default:
 			gr, err := a.resources.GetByID(ctx, *res.ID, apiVersion)
 			if err != nil {
 				a.log.Warn(err) // can happen when the ARM cache is lagging
-				armResources = append(armResources, arm.Resource{
+				a.writeObject(writer, arm.Resource{
 					Resource: res,
 				})
 				continue
 			}
-			armResources = append(armResources, arm.Resource{
+			a.writeObject(writer, arm.Resource{
 				Resource: gr,
 			})
 		}
 	}
 
-	return json.Marshal(armResources)
+	_, err = writer.Write([]byte("]"))
+
+	return err
+}
+
+func (a *azureActions) writeObject(writer io.Writer, resource arm.Resource) {
+	bytes, err := resource.MarshalJSON()
+	if err != nil {
+		a.log.Warn(err) //very unlikely , only a handful of cases trigger an error
+		//here. and since we get the object from a database , it probably will never happen
+	}
+	_, err = writer.Write(bytes)
```

Comment:
if the unlikely error happens above during marshalling should we still attempt to write the bytes?

Reply:
Good catch @s-amann. When I look at the flow again, I would tend to say we keep the flow running. Partial result is better than no result.

Reply:
I agree it would be best not to error out, but the bytes could be empty, nil, or unusable data. My take would be to invert the condition and only write when successful, otherwise log a warning.

Reply:
@s-amann @petrkotas Jeremy gave me push access to his fork and I was able to make the changes. Let me know how it looks.